### ISSUE TITLE
[Bug] Sophisticated Backpacks (Fabric) Incompatiblity

### ISSUE THREAD

**Issue by Xenosentry** [wisp-forest/accessories#20]
If a backpack is worn on a backpack slot, the hotkey for summoning the backpack inventory does not work. Backpack from mod - Sophisticated Backpacks (Unoffical Fabric port). MC 1.20.1 and Fabric 0.15.11
![2024-05-28 165824](https://github.com/wisp-forest/accessories/assets/135584246/5eb1dc98-b597-4f62-a95d-6af5b96c6be6)

**Comment by Xenosentry** [comment]
Addendum to the above.
Also, the on and off hotkeys for the backpack modules do not work. But the modules themselves work, whether it's a magnet that attracts objects or a music block.
![2024-05-28 172832](https://github.com/wisp-forest/accessories/assets/135584246/dd4bafba-70c5-4d98-9430-032e10e30d87)

**Comment by Dragon-Seeker** [comment]
Will investigate such as I currently have another issue where the slot does not work at all.

**Comment by Dragon-Seeker** [comment]
I have tried to investigate issues with this mod but I am having issues running such in production and development. Combined with the fact to which they add trinkets compatibility makes it difficult to add proper support for such meaning it may not be fixed until they deal with such on their side.

**Comment by ChosenMark** [comment]
> If a backpack is worn on a backpack slot, the hotkey for summoning the backpack inventory does not work. Backpack from mod - Sophisticated Backpacks (Unoffical Fabric port). MC 1.20.1 and Fabric 0.15.11 ![2024-05-28 165824](https://private-user-images.githubusercontent.com/135584246/334422019-5eb1dc98-b597-4f62-a95d-6af5b96c6be6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0MTM1NTcsIm5iZiI6MTcxOTQxMzI1NywicGF0aCI6Ii8xMzU1ODQyNDYvMzM0NDIyMDE5LTVlYjFkYzk4LWI1OTctNGY2Mi1hOTVkLTZhZjViOTZjNmJlNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNlQxNDQ3MzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00OWRmYmE1NDkxMGUxYjFlNjc1MDhmMzhlM2JkZmU5NWNlNWEyNDRjYzg2Y2E4ZmRhMzY1YzlmODBiMmM5YWFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.I2z79GwudNKVvL8iGovo-QFxE1wK7P5RC8iicNHNh5g)

same issue for me

**Comment by ChosenMark** [comment]
> If a backpack is worn on a backpack slot, the hotkey for summoning the backpack inventory does not work. Backpack from mod - Sophisticated Backpacks (Unoffical Fabric port). MC 1.20.1 and Fabric 0.15.11

any luck?

**Comment by Salandora** [comment]
So I figured out why this bug happens.

It all comes down to this line: https://github.com/Salandora/SophisticatedBackpacks/blob/c7aa78dac052134b50c71b6a52884e1ed1b5080d/src/main/java/net/p3pp3rf1y/sophisticatedbackpacks/compat/trinkets/TrinketsCompat.java#L96

While trinkets returns a set of TooltipPredicates your compatibility layer does not.
Now I do think this is on my end as it should have been getValidatorPredicates to begin with, so I'll fix this.

That said I think this could still be considered a bug as it does not return the same values as trinkets does.
As further information TrinketInventory and SlotType are both instances of your Wrapped classes.

**Comment by Dragon-Seeker** [comment]
> It all comes down to this line: https://github.com/Salandora/SophisticatedBackpacks/blob/c7aa78dac052134b50c71b6a52884e1ed1b5080d/src/main/java/net/p3pp3rf1y/sophisticatedbackpacks/compat/trinkets/TrinketsCompat.java#L96

~~Ngl that makes little sense why they use tooltip predicates as such should be for tooltip info not logical equipment checks. I could make a patch for this by routing the base predicates to all three method calls instead to fix this issue. Maybe something that can also be changed on the side of Sophisticated Backpacks possible~~

I need to read instead of skimming but I rushed without looking which is my bad, I agree that I should change the behavior of such to return the main set of validators for all three predicates method calls

**Comment by Dragon-Seeker** [comment]
Going to try something in development to see if such is the true problem and adjusting it to the other method call would fix such

**Comment by Dragon-Seeker** [comment]
After making the need changes it still seems to be have a problem which is quite odd... as in my environment I am unable to equip it in any slot.

**Comment by Dragon-Seeker** [comment]
~~Wait I might know what might be going wrong and its to do with the SlotGroup stuff.~~ Sadly did not work and unable to get it working in prod still ):

**Comment by Salandora** [comment]
If it helps I can offer to test your changes in my dev environment. With and without the fix I applied.

**Comment by Dragon-Seeker** [comment]
> If it helps I can offer to test your changes in my dev environment. With and without the fix I applied. 

I pushed the changes to the TCLayer and published such to modrinth and curseforge if u want to test such.

**Comment by Salandora** [comment]
Sorry for the late reply.

The Accessories versions I tested were 1.0.0-alpha.24+1.20, 1.0.0-alpha.26+1.20 and Trinkets Compat Layer 3.8.1+alpha.12 and 3.8.1+alpha.13.
I tested with sophisticated backpacks dev(fixed version) and production(build.104/released version).
And the changes you made work in all versions perfectly for me, I can equip the backpack, open it with the key bind and disable the upgrades with the key bind.
